x-pack/plugins/security_solution/public/management/pages/trusted_apps/view/trusted_apps_list.tsx

```diff
@@ -78,44 +93,61 @@ const getActionDefinitions = (dispatch: Dispatch<Immutable<AppAction>>): Actions
   },
 ];
 
-type ColumnsList = Array<EuiBasicTableColumn<Immutable<TrustedApp>>>;
+const getColumnDefinitions = (context: TrustedAppsListContext): ColumnsList => {
+  const [itemDetailsMap, setItemDetailsMap] = context.detailsMapState;
 
-const getColumnDefinitions = (dispatch: Dispatch<Immutable<AppAction>>): ColumnsList => [
-  {
-    field: 'name',
-    name: COLUMN_TITLES.name,
-  },
-  {
-    field: 'os',
-    name: COLUMN_TITLES.os,
-    render(value: TrustedApp['os'], record: Immutable<TrustedApp>) {
-      return OS_TITLES[value];
+  return [
+    {
+      field: 'name',
+      name: PROPERTY_TITLES.name,
     },
-  },
-  {
-    field: 'created_at',
-    name: COLUMN_TITLES.created_at,
-    render(value: TrustedApp['created_at'], record: Immutable<TrustedApp>) {
-      return (
-        <FormattedDate
-          fieldName={COLUMN_TITLES.created_at}
-          value={value}
-          className="eui-textTruncate"
-        />
-      );
+    {
+      field: 'os',
+      name: PROPERTY_TITLES.os,
+      render(value: TrustedApp['os'], record: Immutable<TrustedApp>) {
+        return OS_TITLES[value];
+      },
     },
-  },
-  {
-    field: 'created_by',
-    name: COLUMN_TITLES.created_by,
-  },
-  {
-    name: COLUMN_TITLES.actions,
-    actions: getActionDefinitions(dispatch),
-  },
-];
+    {
+      field: 'created_at',
+      name: PROPERTY_TITLES.created_at,
+      render(value: TrustedApp['created_at'], record: Immutable<TrustedApp>) {
+        return (
+          <FormattedDate
+            fieldName={PROPERTY_TITLES.created_at}
+            value={value}
+            className="eui-textTruncate"
+          />
+        );
+      },
+    },
+    {
+      field: 'created_by',
+      name: PROPERTY_TITLES.created_by,
+    },
+    {
+      name: ACTIONS_COLUMN_TITLE,
+      actions: getActionDefinitions(context),
+    },
+    {
+      align: RIGHT_ALIGNMENT,
+      width: '40px',
+      isExpander: true,
+      render(item: Immutable<TrustedApp>) {
+        return (
+          <EuiButtonIcon
+            onClick={() => setItemDetailsMap(toggleItemDetailsInMap(itemDetailsMap, item, context))}
+            aria-label={itemDetailsMap[item.id] ? 'Collapse' : 'Expand'}
+            iconType={itemDetailsMap[item.id] ? 'arrowUp' : 'arrowDown'}
+          />
+        );
+      },
+    },
+  ];
+};
 
 export const TrustedAppsList = memo(() => {
+  const [detailsMap, setDetailsMap] = useState<DetailsMap>({});
```

Comment:
nice way to still keep the `useState` in the same place as the selectors.  could make it easy to switch out if we ever needed to